### ISSUE TITLE
Confidence on training NLU model is 0 for most intents

### ISSUE THREAD

**Issue by karan3194** [RasaHQ/rasa#3383]
**Rasa version**: rasa-core                     0.14.1   
                            rasa-core-sdk                0.14.0   
                            rasa-nlu                      0.15.0   

**Python version**: 3.6.8

**Operating system** (windows, osx, ...): Linux 18.04.1

**Issue**:
I'm building a model which allows users' to be able to check the price of various airlines through the chatbot I am training my RASA NLU model on 5 intents and proving some training data for each. However, the confidence score for many of the intents post NLU training is still 0. Is there something I am doing wrong?  

**Content of configuration file (config.yml)**:

language: "en"

pipeline:
- name : "nlp_spacy"
- name: "tokenizer_whitespace"
- name: "intent_entity_featurizer_regex"
- name: "ner_crf"
- name: "ner_synonyms"
- name: "intent_featurizer_count_vectors"
- name: "intent_classifier_tensorflow_embedding"
  intent_tokenization_flag: true
intent_split_symbol: "+" `

**Content of domain file (domain.yml)** (if used & relevant):
````intents:
- greeting
- flight
- inform
- affirmation
- deny

entities:
- location 
- date

slots:
   from:
            type: text
   to:
            type: text
   date:
            type: text

actions:
- utter_greeting
- utter_boarding
- utter_destination
- utter_date
- utter_confirm
- utter_check_another_one
- utter_thanks
- utter_unclear
- action_save_origin
- action_save_destination
- action_save_date
- action_get_flight
- action_slot_reset

templates:  
  utter_greeting:
  - text: "hello!"

  utter_boarding:
  - text: "We'll help you find the latest flight schedule. First, please provide your origin airport code?"
  
  utter_destination:
  - text: "And the destination airport code?"
  
  utter_date:
  - text: "What is the date for your travel(in dd-mm-yyyy)?"
  
  utter_confirm:
  - text: "I will be making inquiry for flight from {from} to {to} on {date}. Is that correct?"
  
  utter_check_another_one:
  - text: "Do you want to make another inquiry?"
  
  utter_thanks:
  - text: "Thanks for contacting us. Have a good day!"
  
  utter_unclear:
  - text: "I am not sure what you are aiming for.Kindly try it again"
  
**Content of NLU data file (nlu.md)**
```
## intent:greeting
-hi
-hello
-goodmorning
-goodafternoon
-hey
-whatsup?
-greetings
-how are you
-whats up
-how are you doing
-how is it going
-good morning
-good afternoon
-good day
-good night
-are you there
-morning
-how are things going
-are you around
-whatsup
-sup
-are you around?
-greetings
-Hi, How is it going?
-Nice to meet you.
-How do you do?
-Hi, nice to meet you.
-It is a pleasure to meet you.
-Top of the morning to you!
-What's up?


## intent:flight
-hey! I want latest flight schedule
-can you provide me flight schedule
-i want the flight schedule
-i want to know the flight schedule
-what is the flight schedule?
-give me some flight information
-i need information about a few flights
-i need information on a flight
-i need to know about a flight
-can you tell me fligh prices?
-please tell me flight prices
-please provide latest flight info
-is it possible to get latest flight information
-flight schedule
-flight information
-flight info

## intent:inform
-[DEL](location)
-[BOM](location)
-[CCU](location)
-[BLR](location)
-[HYD](location)
-[MAA](location)
-[PNQ](location)
-[IXC](location)
-[09-01-2019](date)
-[20-05-2019] (date)
- I want for [09-02-2019](date)
- i want to travel on [21-04-2019](date)


## intent:affirmation
-ok
-yeah
-yes
-definitely
-ya
-yup
-exactly
-right
-definitely
-absolutely
-certainly
-exactly
-right
-affirmative
-yes to that
-correct
-yep
-yes sir
-si
-correcto
-allright
-i agree
-that is correct
-that is right
-you got that
-yes that is right
-yup that is correct
-yeah that is right
-sounds good


## intent:deny
-no
-not yet
-never
-not now
-nope
-nah
-na
-wrong
-negative
-not correct
-incorrect
-no sir
-dont agree on that
-didnt meant that
-dont mean that
-not what i want
-not correct
-bad
-im not ok with that
-not good
-that is wrong
-im not good with that
-ok its not fine
-definitely no
-not good for me
-its not ok
-its not correct

**Content of CORE data file (stories.md)**

## fallback
- utter_unclear

## Story 1
* greeting
    -utter_greeting
* flight
    - utter_boarding
* inform{"location": "BOM"}
    - action_save_origin
    - slot{"from": "BOM"}
    - utter_destination
* inform{"location": "DEL"}
    - action_save_destination
    - slot{"to": "DEL"}
    - utter_date
* inform{"date": "20-01-2019"}
    - action_save_date
    - slot{"date": "20-01-2019"}
    - utter_confirm
* affirmation
    - action_get_flight
	- utter_check_another_one
* deny
	- utter_thanks
	- action_restart

## Story 2-multiple steps
* greeting
    -utter_greeting
* flight
    - utter_boarding
* inform{"location": "CCU"}
    - action_save_origin
    - slot{"from": "CCU"}
    - utter_destination
* inform{"location": "BLR"}
    - action_save_destination
    - slot{"to": "BLR"}
    - utter_date
* inform{"date": "03-02-2019"}
    - slot{"date": "03-02-2019"}
    - action_save_date
    - slot{"date": "03-02-2019"}
    - utter_confirm
* affirmation
    - action_get_flight
    - utter_check_another_one
* affirmation
	- action_slot_reset
	- reset_slots
    - utter_boarding
* inform{"location": "BOM"}
    - action_save_origin
    - slot{"from": "BOM"}
    - utter_destination
* inform{"location": "DEL"}
    - action_save_destination
    - slot{"to": "DEL"}
    - utter_date
* inform{"date": "10-02-2019"}
    - slot{"date": "10-02-2019"}
    - action_save_date
    - slot{"date": "10-02-2019"}
    - utter_confirm
* affirmation
    - action_get_flight
    - utter_check_another_one
* deny
    - utter_thanks
    - action_restart

**Comment by akelad** [comment]
Thanks for raising this issue, @Ghostvv  will get back to you about it soon.

**Comment by Ghostvv** [comment]
even those from training data?

**Comment by karan3194** [comment]
Solved this issue.. I'm facing another problem now. I am trying the Berlin Rasa Workshop project uploaded on your Git. (https://github.com/RasaHQ/rasa-workshop-pydata-berlin)
The issue that I'm facing is that while testing my NLU model, it is supposed to show the performance metrics for F1 score, precision, recall and a confusion matrix too, instead it shows everything in the form of a Dictionary/Json format making it easy to interpret the output.
I've attached screenshots for reference. (RASA.png is my implementation while the other is from the Berlin workshop notebook)
Do i need to convert the dictionary to some other format and then plot it? TIA.
![rasa_berlin](https://user-images.githubusercontent.com/50234516/57358304-c3768880-7192-11e9-9352-98d32d9fd50a.png)
![rasa](https://user-images.githubusercontent.com/50234516/57358305-c3768880-7192-11e9-8475-6681165c6e91.png)
